pkg/cloudprovider/azure/common.go

```diff
@@ -63,21 +64,33 @@ func loadEnv() error {
 	return nil
 }
 
-func newServicePrincipalToken(tenantID, clientID, clientSecret, scope string) (*adal.ServicePrincipalToken, error) {
-	oauthConfig, err := adal.NewOAuthConfig(azure.PublicCloud.ActiveDirectoryEndpoint, tenantID)
+// ParseAzureEnvironment returns azure environment by name
+func parseAzureEnvironment(cloudName string) (*azure.Environment, error) {
+	var env azure.Environment
+	var err error
+	if cloudName == "" {
+		env = azure.PublicCloud
+	} else {
+		env, err = azure.EnvironmentFromName(cloudName)
+	}
+	return &env, err
+}
+
+func newServicePrincipalToken(tenantID, clientID, clientSecret string, env *azure.Environment) (*adal.ServicePrincipalToken, error) {
+	oauthConfig, err := adal.NewOAuthConfig(env.ActiveDirectoryEndpoint, tenantID)
 	if err != nil {
 		return nil, errors.Wrap(err, "error getting OAuthConfig")
 	}
 
-	return adal.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, scope)
+	return adal.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, env.ResourceManagerEndpoint)
 }
 
 func getRequiredValues(getValue func(string) string, keys ...string) (map[string]string, error) {
 	missing := []string{}
 	results := map[string]string{}
 
 	for _, key := range keys {
-		if val := getValue(key); val == "" {
+		if val := getValue(key); val == "" && key != cloudNameEnvVar {
```

Comment:
Let's remove this special case, since the cloud name is **not** a required value. see next comment.